backend/clubs/views.py

```diff
@@ -2878,33 +2878,64 @@ def get(self, request):
-        ans = conn.get(key)
-        if ans is None:
-            ans = 0
-        else:
-            ans = int(ans)
-        return Response({"count": ans})
+        meeting_id = request.query_params.get("event")
+        return Response({"count": ZoomMeetingVisit.objects.filter(meeting_id=meeting_id).count()})
 
     def post(self, request):
         action = request.data.get("event")
-        meeting_id = request.data.get("payload", {}).get("object", {}).get("id", None)
-        if meeting_id is not None:
-            conn = get_redis_connection()
-            key = f"zoom:meeting:live:{meeting_id}"
-            if action == "meeting.participant_joined":
-                conn.incr(key)
-            elif action == "meeting.participant_left":
-                conn.decr(key)
-            conn.expire(key, datetime.timedelta(hours=8))
+        if action == "meeting.participant_joined":
+            email = (
+                request.data.get("payload", {})
+                .get("object", {})
+                .get("participant", {})
+                .get("email", None)
+            )
+            person = get_user_model().objects.filter(email=email)
+
+            event_name = request.data.get("payload", {}).get("object", {}).get("topic", None)
+            event = Event.objects.filter(name=event_name).order_by("-start_time")[0]
+
+            meeting_id = request.data.get("payload", {}).get("object", {}).get("id", None)
+            participant_id = (
+                request.data.get("payload", {})
+                .get("object", {})
+                .get("participant", {})
+                .get("user_id", None)
+            )
+            join_time = (
+                request.data.get("payload", {})
+                .get("object", {})
+                .get("participant", {})
+                .get("join_time", None)
+            )
+
+            if event:
+                club = event.club
+                ZoomMeetingVisit.objects.create(
+                    person=person,
+                    club=club,
+                    event=event,
+                    meeting_id=meeting_id,
+                    participant_id=participant_id,
+                    join_time=join_time,
+                )
+        elif action == "meeting.participant_left":
+            meeting_id = request.data.get("payload", {}).get("object", {}).get("id", None)
+            participant_id = (
+                request.data.get("payload", {})
+                .get("object", {})
+                .get("participant", {})
+                .get("user_id", None)
+            )
+            leave_time = (
+                request.data.get("payload", {})
+                .get("object", {})
+                .get("participant", {})
+                .get("leave_time", None)
+            )
+            ZoomMeetingVisit.objects.filter(meeting_id=meeting_id).filter(
+                participant_id=participant_id
+            ).update(leave_time=leave_time)
```

Comment:
This won't work if the user joins and leaves the meeting multiple times, you'll set all of the leave times to be the latest value.

There is also another edge case where you have the user join the meeting twice from two different devices. Although this is a rare case, it would be nice to handle it.

Reply:
You can also combine a `.filter(a).filter(b)` into a `.filter(a, b)`. When you do comma separated, it is `and` by default. If you want to do `or`, it would be `.filter(Q(a) | Q(b))`.